### DISCUSSION TITLE
How to use it on Server/API

### DISCUSSION THREAD

**Discussion by casualmatt** [Intevel/nuxt-directus#22] [Q&A] [answered]
Hello!

Is somebody able to give me some tips on how to use the module on a Server/API?
i want to automate some flow/action and i would need to access the DB from server api, and not just form front end. 
For now i get an error on the console if i try to import the function:
```
[nuxt] [request error] useDirectusItems is not defined
  at ./.nuxt/dev/index.mjs:332:24  
  at processTicksAndRejections (node:internal/process/task_queues:96:5)  
  at async ./node_modules/h3/dist/index.mjs:416:19  
  at async Server.nodeHandler (./node_modules/h3/dist/index.mjs:366:7)
```

**Comment by Intevel** [comment, marked as answer]
I think there is no way to use Nuxt Modules in the Server.

**Reply by genu** [reply]
I realize this is a pretty old answer, however, it's not correct.

Nuxt modules can provide server utilities. See: https://github.com/nuxt-modules/supabase

Specifically, the `runtime/server` folder. A directus service can be defined there and it would be automatically imported for the user to consume.

**Reply by Intevel** [reply]
We can write on composables for that, because currently we have `useCookie` used in our composables, which not work on Server Side.

**Reply by AnmolBansalDEV** [reply]
is there a workaround to use the module in the server?

**Reply by Intevel** [reply]
> is there a workaround to use the module in the server?
> 
> 

No, currently not

**Reply by genu** [reply]
what you can do, (its what I ended up doing), Is just use the official directus js API on the server. You can do this by wrapping it in a utility and just import it. Kind of like this:

I'm using the `@directus/sdk` beta here, but you can use the stable one:

server/services/directus.ts:
```ts
/**
 * We alias `useDirectus` to `directus` to avoid conflict with the `useDirectus` composible provided by `nuxt-directus`.
 */
import { useDirectus as directus, rest, readItem, readItems, authentication } from '@directus/sdk'

interface Schema {
  // ... your schema here. You can reuse the types created for `nuxt-directus`
}

/**
 * Configure the env variables in `nuxt.config.ts` and use them here.
 * directusUrl: process.env.NUXT_PUBLIC_DIRECTUS_URL,
 * directusStaticToken: process.env.NUXT_DIRECTUS_STATIC_TOKEN,
 */
const {
  public: { directusUrl },
  directusStaticToken,
} = useRuntimeConfig()

/**
 * We create two instances of the client. One for the service role, and one for the client role.
 * The Service role uses the static token to access the API with unrestricted permissions, and the
 * client role uses the JWT token.
 */
const serviceRoleClient = directus<Schema>(directusUrl).use(rest()).use(authentication('json'))
const client = directus<Schema>(directusUrl).use(rest()).use(authentication('json'))

serviceRoleClient.setToken(directusStaticToken)

export const useDirectusClient = () => {
  return { serviceRoleClient, client, readItem, readItems }
}
```

To use this service:
```ts
import { useDirectusClient } from '../services/directus.ts';

export default defineEventHandler(async (event) => {
const { readItems, serviceRoleClient } = useDirectusClient()

/**
 * You should have type safety here
 */
const customers = await serviceRoleClient.request(readItems('customers'), { fields: ['*'] })

console.log(customers)
}
```
